File: Work/WK/MS/Management/Management/Management/Management.Web/Modules/ServerTypes/Administration/UserForm.ts
import { StringEditor, LookupEditor, PasswordEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface UserForm {
    Username: StringEditor;
    Roles: LookupEditor;
    Password: PasswordEditor;
    PasswordConfirm: PasswordEditor;
}

export class UserForm extends PrefixedContext {
    static formKey = 'Administration.User';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!UserForm.init)  {
            UserForm.init = true;

            var w0 = StringEditor;
            var w1 = LookupEditor;
            var w2 = PasswordEditor;

            initFormType(UserForm, [
                'Username', w0,
                'Roles', w1,
                'Password', w2,
                'PasswordConfirm', w2
            ]);
        }
    }
}

File: Work/WK/MS/Management/Management/Management/Management.Web/Modules/Administration/User/UserForm.cs
using Serenity.ComponentModel;
using System.Collections.Generic;

namespace Management.Administration.Forms
{
    [FormScript("Administration.User")]
    [BasedOnRow(typeof(UserRow), CheckNames = true)]
    public class UserForm
    {
        [LabelWidth(200, UntilNext = true)]
        public string Username { get; set; }

        [LookupEditor(typeof(RoleRow), Multiple = true)]
        public List<int> Roles { get; set; }
     
        [PasswordEditor, Required(true)]
        public string Password { get; set; }
        [PasswordEditor, OneWay, Required(true)]
        public string PasswordConfirm { get; set; }
      
    }
}